notebooks/ch-mbqc/measurement.ipynb

```diff
@@ -0,0 +1,497 @@
+    "\n",
+    "In our case, we also need to apply the same correction concept which is based on what is known as Pauli stablizers.\n",
+    "Simply put, Pauli stablizers are set of operators, each associated to one vertex of the graph, the idea is that a stablizer $K$ should not change the graph $|G\\rangle$ so\n",
+    "$$ K_i := X_i \\otimes_ {j\\in N(i)} Z_j $$\n",
+    "$$ K_i |G\\rangle = |G\\rangle $$\n",
+    "\n",
+    "\n",
+    "The Flow tells us how to apply the stabilisers to correct side-effects of projective measurements. \n",
+    "\n",
+    "Measuring in Z-basis (computational basis) is equivalent to cutting the node from the graph and if the measurement outcome is 1, then a Z gate gets applied to all of its neighbors (nodes it had a cz gate with) in the remaining umeasured part of the cluster state. \n",
+    "\n",
+    "Measuring in X-basis (i.e applying a Hadamard gate before measuring in the computational basis) gives an interesting effect that can be illustrated as follows:\n",
+    "\n",
+    "When measuring a node $n_i$, if measurement outcome of $n_i =1$:\n",
+    "- Apply X correction to the successor (according to flow, usually the node in the next layer on the same row)\n",
+    "- Apply Z correction to all neighbors of the successor (excluding $n_i$)\n",
+    "\n",
+    "if measurement outcome of $n_i = 0$ we do nothing.\n",
+    "\n",
+    "Another way of saying this is: In MBQC (initialized with $|+\\rangle$ state) we always want to obtain a 0, if we do not get a zero, it means an error occured and a correction needs to follow.\n",
+    "\n",
+    "An illustration of this for a 2D cluster when we measure qubit 1 and get 1:\n",
+    "\n",
+    "![Byproduction correction illustration for 1st qubit](images/measure-1.png)\n",
+    "\n",
+    "Then we measure qubit 2 and get 1:\n",
+    "\n",
+    "![Byproduction correction illustration for 2nd qubit](images/measure-2.png)\n",
+    "\n",
+    "This simple flow we used here works fine for brickwork and cluster states, for other irregular graph shapes, if a flow exists, there are ways to find it and find the order of measurements and corrections, but it is beyond the scope of this guide.\n",
+    "\n",
+    "For the linear cluster example we are trying to implement:\n",
+    "\n",
+    "![Byproduction correction illustration for linear cluster](images/linear-correction.png)\n",
+    "\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Implementation of measurement\n",
+    "\n",
+    "First of all, we are going to defind a function for X measurement, also instead of using the usual Qiskit notation with LSB first, we are going to use MSB first for the sake of easier readability and comparaison with the visual cluster"
```

Comment:
```suggestion
    "First of all, we are going to define a function for X measurement, also instead of using the usual Qiskit notation with LSB first, we are going to use MSB first for the sake of easier readability and comparison with the visual cluster"
```